cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/upgrade/AbstractConfigurationUpgrader.java

```diff
@@ -192,32 +260,60 @@ private VERSIONS_TYPE resolveLatestCompatibleSchemaVersion(Path configurationRoo
    */
   public boolean upgradeConfigurationToLatestVersion(Path configurationRoot, BackupPolicy backupPolicy) {
 
+    return upgradeConfigurationToLatestVersion(configurationRoot, backupPolicy, null);
+  }
+
+  /**
+   * Upgrades the configuration to the latest supported version with option to limit lookup of the maximum version.
+   *
+   * @param configurationRoot the root folder containing the configuration with the specified
+   *        {@link #configurationFilename}. See {@link #AbstractConfigurationUpgrader(Enum, Class, String)} for more
+   *        information.
+   * @param backupPolicy the {@link BackupPolicy} to choose if a backup is necessary or not.
+   * @param maxVersion the latest version to be used for the upgrade (limits the versions to choose from)
+   * @return if manual adoptions has to be performed after upgrading
```

Comment:
manual adoption has to be... or manual adoptions have to be...?

Reply:
I think "has"

Reply:
you can either use: manual adoptions have to be performed...
or: if a manual adoption has to be performed
but not: manual adoptions has to be performed.